l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -189,18 +192,14 @@ library Decoder {
     pure
     returns (bytes32)
   {
-    bytes memory temp = new bytes(0x120);
+    // 0x20 for the block number + 0x11c for the header elements
+    bytes memory temp = new bytes(0x20 + 0x11c);
 
     assembly {
-      // Copy the L2 block number byte by byte to memory
-      mstore8(add(temp, 0x20), shr(24, _l2BlockNumber))
-      mstore8(add(temp, 0x21), shr(16, _l2BlockNumber))
-      mstore8(add(temp, 0x22), shr(8, _l2BlockNumber))
-      mstore8(add(temp, 0x23), _l2BlockNumber)
-    }
-    assembly {
-      // Copy header elements (not including block number) for start or end (size 0xd4)
-      calldatacopy(add(temp, 0x24), add(_l2Block.offset, _offset), 0x11c)
+      // Copy block number
+      mstore(add(temp, 0x20), _l2BlockNumber)
```

Comment:
I don't think I get this. To copy block number we should be doing `mstore(0x40)` right? But here we are doing `0x20+0x11c+020`

Reply:
I don't get what you are saying. Are you pointing to the correct place in the code? Where are we doing `0x20+0x11c+0x20`?

Line 200 is simply copying the `_l2BlockNumber` provided, and putting it as the first 32 usable bytes in the `temp` (we skip 32 bytes as the length is stored there).

Line 202 is copying 0x11c bytes from the l2 block into the temp starting just after the block number. These bytes are state trees from the header.

Reply:
ah i see nvm got confused lol